utils/build_ds_container.py

```diff
@@ -0,0 +1,257 @@
+    help=(
+        'Build content in-cluster. Note that this option ignores the '
+        '--product and --debug flags.'),
+    action='store_true',
+    default=False)
+parser.add_argument(
+    '-d', '--debug',
+    help=(
+        'Provide debug output during the build process. This option is '
+        'ignored when building content in the cluster using '
+        '--build-in-cluster.'),
+    action='store_true',
+    default=False)
+parser.add_argument(
+    '-P', '--product',
+    help=(
+        'The product(s) to build. This option can be provided multiple times. '
+        'This option is ignored when building content in the cluster using '
+        '--build-in-cluster.'),
+    default=['ocp4', 'rhcos4'],
+    dest='products',
+    nargs='*')
+parser.add_argument(
+    '-r', '--repository',
+    help=(
+        'The container image repository to use for images containing built '
+        'content. Images pushed to this repository must have a tag, which '
+        'you can specify with the --container-image-tag argument. If you do '
+        'not supply a tag, one will be generated for you. It is recommended '
+        'that you properly tag built images for production use. '
+        'Auto-generated tags are primarily useful for development workflows. '
+        'This script assumes you have authenticated to the image repository '
+        'if necessary (e.g, `podman login`).'))
+parser.add_argument(
+    '-t', '--container-image-tag',
+    help='A unique tag for the container image.')
+args = parser.parse_args()
+
+
+log.basicConfig(
+    format='%(asctime)s:%(levelname)s: %(message)s', level=log.INFO)
+
+
+# FIXME(lbragstad): Remove this and replace it with operating system utils if
+# possible.
+result = subprocess.run(
+    ['git', 'rev-parse', '--show-toplevel'], capture_output=True, check=True)
+# Convert the file path from bytes to unicode since we might manipulate it
+# later. Also, strip off any newlines.
+REPO_PATH = result.stdout.decode().strip()
+
+
+def ensure_namespace_exists(namespace):
+    """Function that ensures there is a namespace for the content."""
+    with client.ApiClient() as api_client:
+        body = client.V1Namespace(
+            api_version='v1', kind='Namespace', metadata={'name': namespace})
+
+        try:
+            api = client.CoreV1Api(api_client)
+            api.create_namespace(body)
+            log.info('Created namespace %s', namespace)
+        except exceptions.ApiException as err:
+            if err.status == 409:
+                log.info('Reusing existing namespace: %s', namespace)
+            else:
+                raise err
+
+
+def supports_custom_resource(resource_name):
+    with client.ApiClient() as api_client:
+        api = client.ApiextensionsV1Api(api_client)
+        try:
+            api.read_custom_resource_definition(resource_name)
+            return True
+        except exceptions.ApiException as err:
+            return False
+            if err.status == 404:
```

Comment:
Hmm, can we reach this if the previous line is `return False` ?

Reply:
Yep - good catch. I didn't finish smoothing out the wrinkles here before pushing this version.

I removed the if statement since I don't think it matters. Instead, I just logged something for the user and returned false.